ddtrace/tracer/spancontext.go

```diff
@@ -179,6 +191,22 @@ func (t *trace) setSamplingPriority(p float64) {
 	t.setSamplingPriorityLocked(p)
 }
 
+func (t *trace) keep() {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	if t.samplingDecision == decisionDefaultDrop {
+		t.samplingDecision = decisionForceKeep
+	}
+}
+
+func (t *trace) drop() {
+	t.mu.Lock()
+	defer t.mu.Unlock()
```

Comment:
I think you can avoid locking here by doing something ugly like:
```go
atomic.StoreInt64((*int64)(&t.samplingDecision), int64(decisionDrop))
```
And also
```
samplingDecision(atomic.LoadInt64((*int64)(&t.samplingDecision)))
```

Reply:
Makes sense. I went for compareAndSwap.